Calculator_program.c

```diff
@@ -0,0 +1,127 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#define MAX_LEN 100 // Maximum length
+
+// Function to calculate expressions 
+float calculateExpression(char *expr) {
+    float values[MAX_LEN]; 
+    char operators[MAX_LEN]; 
+    int valTop = -1; 
+    int opTop = -1; 
+
+    int i = 0;
+    //Function to perform operations
+    float applyOperation(float a, float b, char op) {
+        if (op == '+') return a + b;
+        if (op == '-') return a - b;
+        if (op == '*') return a * b;
+        if (op == '/') {
+            if (b == 0) {
+                printf("Division by zero is not allowed.\n");
+                exit(1);
+            }
+            return a / b;
+        }
+        return 0;
+    }
+
+    // Function to follow precedence of operators
+    int precedence(char op) {
+        if (op == '+' || op == '-') return 1; // Lowest precedence
+        if (op == '*' || op == '/') return 2; // Higher precedence
+        return 0;
+    }
+
+    // Function Process the expression
+    while (expr[i] != '\0') {
+        char c = expr[i];
+
+        // Skip spaces
+        if (c == ' ') {
+            i++;
+            continue;
+        }
+
+        // If the character is a number, parse it
+        if (isdigit(c) || c == '.') {

```

Comment:
Instead of using built-in functions like isdigit(), Can you create your own definition of these functions and remove unnecessary headerfiles?

Reply:
👍